### ISSUE TITLE
[Bug] There is no smart hint in VS by the extension tag defined by IMarkupExtension.

### ISSUE THREAD

**Issue by MackYang** [xamarin/Xamarin.Forms#6678]
### Description
I want to create a style you can apply multiple tags simultaneously (as in HTML using CSS like), I defined, can be used normally, but not smart tips in VS, unlike StaticResource as when you are prompted to press the space available Style Key

我想创建一个可以同时应用多个样式的标记(像HTML中使用CSS那样),我定义完成后,可以正常使用,但是在VS中没有智能提示,不像StaticResource那样在按空格时会提示可用的Style Key
### Steps to Reproduce

1. create markup
`cs

using MutliStyleDemo.Code;
using System;
using System.Linq;
using System.Reflection;
using System.Xml;
using Xamarin.Forms;
using Xamarin.Forms.Xaml;
using XamlParseException = MutliStyleDemo.Code.XamlParseException;

namespace MutliStyleDemo
{
    [ContentProperty("Key")]
    public class MultiStyleExtension :IMarkupExtension
    {
        public string Key { get; set; }
        private string[] resourceKeys;
        public object ProvideValue(IServiceProvider serviceProvider)
        {
            this.resourceKeys = Key.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
            Style resultStyle = new StaticResourceExtension() { Key = resourceKeys[0] }.ProvideValue(serviceProvider) as Style;

            for (int i = 1; i < resourceKeys.Length; i++)
            {
                var currentResourceKey = resourceKeys[i];
                Style currentStyle = new StaticResourceExtension() { Key = currentResourceKey }.ProvideValue(serviceProvider) as Style;

                if (currentStyle == null)
                {
                    throw new InvalidOperationException("Could not find style with resource key " + currentResourceKey + ".");
                }

                resultStyle.Merge(currentStyle);

            }

            return resultStyle;
        }

    }

    public static class Extension
    {
        public static void Merge(this Style style1, Style style2)
        {
            if (style1 == null)
            {
                throw new ArgumentNullException("style1");
            }
            if (style2 == null)
            {
                throw new ArgumentNullException("style2");
            }

            if (style2.BasedOn != null)
            {
                Merge(style1, style2.BasedOn);
            }

            foreach (var currentSetter in style2.Setters)
            {
                style1.Setters.Add(currentSetter);
            }

            foreach (TriggerBase currentTrigger in style2.Triggers)
            {
                style1.Triggers.Add(currentTrigger);
            }
        }
    }
}

`
2. 
`xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:local="clr-namespace:MutliStyleDemo"
             x:Class="MutliStyleDemo.MainPage">
    <ContentPage.Resources>
        <Style TargetType="Label" x:Key="s1">
            <Setter Property="TextColor" Value="Accent"/>
        </Style>

        <Style TargetType="Label" x:Key="s2">
            <Setter Property="FontSize" Value="20"/>
        </Style>

        <Style TargetType="Label" x:Key="s3">
            <Setter Property="FontAttributes" Value="Italic"/>
        </Style>
      
    </ContentPage.Resources>

 
    
    <StackLayout>
        <Label Text="Welcome to Xamarin.Forms!"   Style="{local:MultiStyle s1 s2 }"
           HorizontalOptions="Center"
           VerticalOptions="CenterAndExpand" />

        <Label Text="Welcome to Xamarin.Forms!"  Style="{StaticResource s2 }"
           HorizontalOptions="Center"
           VerticalOptions="CenterAndExpand" />
    </StackLayout>

</ContentPage>

`

3. 

### Expected Behavior
I hope that when I press the space, I will have a smart prompt, like the one below.
![image](https://user-images.githubusercontent.com/24520545/60245088-5bb90000-98ee-11e9-8d45-e6516ab12407.png)

### Actual Behavior
But when I use custom Markups, I don't have smart tips, No styles are listed,like the one below.
![image](https://user-images.githubusercontent.com/24520545/60245186-a175c880-98ee-11e9-8a68-3f90b34534b0.png)

### Basic Information

- Version with issue:
- Last known good version:
- IDE:VS2019 16.00
- Platform Target Frameworks:  
XamarinForm
- Nuget Packages:
- Affected Devices:

**Comment by jsuarezruiz** [comment]
I have done tests with a custom extension and effectively, there is no intellisense in this case.
[Test6678.zip](https://github.com/xamarin/Xamarin.Forms/files/3406507/Test6678.zip)

![issue6678](https://user-images.githubusercontent.com/6755973/61458420-bb914c80-a96a-11e9-83fe-61488407082a.gif)

**Comment by groege** [comment]
I also think this would be a good feature --> especially for *.resx files
Additionally a compile error or warning that tells you that a value is not available (imagine you say {i18n:Translate Example} but due to a typo {i18n:Tranlate} the string is empty...)

**Comment by samhouts** [comment]
cc @mgoertz-msft

**Comment by mgoertz-msft** [comment]
That's correct, for custom markup extensions there is no IntelliSense beyond the list of properties of the the markup extension. The language service has no idea what the values of the properties for custom markup extensions should be unless they are simple types, such as enums or bool, with a known set of possible values.
